Introduces support for instantiating packages from remote project links

hennevogel opened the pull request:
Instead of returning nil for packages that might come from an remote project link, check if they are known to the backend and return an in-memory, read only Package object.

hennevogel commented:
Deployed on build-test. 

https://build-test.opensuse.org/project/show/home:hennevogel

hennevogel commented:
TODO: 

- [ ] Project#index should list the packages in an "Inherited Packages" tab somehow...
- [ ] Branching a package leads to a weird package because the `BackendPackage` does not consider `linked` information from the backend
- [ ]

hennevogel commented:
Do not merge this. 

It needs some more sophisticated authorization setup. We probably want to switch the default of the option and then explicitly enable this per `Package.get_by_project_and_name` call and figure out problems first....

hennevogel commented:
I switched it off by default so we can experiment with it after merging.

krauselukas reviewed (commented):
@hennevogel `test/functional/source_controller_test.rb:2344` is failing. Other then that it looks good to me :)

hennevogel commented:
Yeah it's failing because of the non-sensical package cache. Looks like I have to kill this first...

hennevogel commented:
Starting here #16519

hennevogel committed:
Introduces support for instantiating packages from remote project links

Instead of returning nil for packages that might come from an remote project
link, check if they are known to the backend and return an in-memory, read only
Package object.

hennevogel committed:
Spec out instantiating packages from remote project links

hennevogel commented:
@krauselukas this should be it, if you want to have a look again
